File: ikernell-frontend/src/types/interrupcion.ts
import type { ActividadResponse } from './actividad';
import type { TipoInterrupcionResponse, UsuarioResumenResponse } from './usuario';

export interface InterrupcionResponse {
  idInterrupcion: number;
  codigoInterrupcion: string;
  actividad: ActividadResponse;
  tipoInterrupcion: TipoInterrupcionResponse;
  /** NUEVO: quién la registró -- hoy siempre el desarrollador de la actividad. */
  usuarioCreador: UsuarioResumenResponse | null;
  motivo: string;
  duracionMinutos: number;
  fechaRegistro: string;
}

/** Igual que RegistroError: sin campo "fase" propio, se deriva de actividad.etapa. */
export interface InterrupcionRequest {
  idActividad: number;
  idTipoInterrupcion: number;
  motivo: string;
  duracionMinutos: number;
}

File: ikernell-backend/src/main/java/com/ikernell/backend/entity/Interrupcion.java
package com.ikernell.backend.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;

@Entity
@Table(name = "interrupcion")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Interrupcion {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_interrupcion")
    private Integer idInterrupcion;

    @Column(name = "codigo_interrupcion", nullable = false, length = 20, unique = true)
    private String codigoInterrupcion;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "id_actividad", nullable = false)
    private Actividad actividad;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "id_tipo_interrupcion", nullable = false)
    private TipoInterrupcion tipoInterrupcion;

    /** NUEVO: quién la registró -- hoy siempre el desarrollador de la actividad. */
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "id_usuario_creador")
    private Usuario usuarioCreador;

    @Column(name = "motivo", nullable = false, columnDefinition = "TEXT")
    private String motivo;

    @Column(name = "duracion_minutos", nullable = false)
    private Integer duracionMinutos;

    @Column(name = "fecha_registro", nullable = false, updatable = false)
    @Builder.Default
    private LocalDateTime fechaRegistro = LocalDateTime.now();
}
